rust-mode.el

```diff
@@ -96,6 +96,13 @@
     )
    ))
 
+;;; Start of a Rust item
+(defvar rust-top-item-beg-re
+  (concat "\\s-*\\(?:priv\\|pub\\)?\\s-*"
```

Comment:
Yeah, nice catch -- if I wrap the new test case in an outer `fn`, indentation fails once again.